docs/LoRA.md

```diff
@@ -121,72 +170,151 @@ if config.merged_hf_save_path is not None:
 ```
 
 
-#### Using the checkpoints in Hugging Face's PEFT library
+## The Configuration File
 
-You can use the checkpoints in Hugging Face's PEFT library by doing something like this:
+Here's the complete configuration file for the model.
 
-```python
-peft_config = PeftConfig.from_pretrained(path)
-base_model = AutoModelForCausalLM.from_pretrained(peft_config.base_model_name_or_path)
-lora_model = PeftModel.from_pretrained(base_model, path)
+```yaml
+model_name_or_path: "meta-llama/Llama-2-7b-hf"
+data: gsm8k
+trainer:
+  mp: p=f32,c=bfloat16
+  wandb:
+    project: "levanter-gsm8k"
+    tags: ["gsm8k", "lora", "llama2"]
+  num_train_steps: 550  # 64 * 550 = 35200, which is a bit more than 4 epochs
+  train_batch_size: 64
+
+  # if using model parallelism, this is useful:
+  tensor_parallel_axes: ["mlp", "heads"]
+optimizer:
+  # values in qlora paper
+  learning_rate: 2e-4
+  weight_decay: 0.0
+  lr_schedule: "constant"
+lora:
+  # These are the defaults, but just so you can see them
+  r: 8  # rank of LoRA transform
+  alpha: 8.0  # scaling factor for LoRA transform
+  dropout: 0.0  # dropout probability for LoRA layers
 ```
 
-#### Using the merged checkpoints
 
-You can use the merged checkpoints with Hugging Face's inference scripts by doing something like this:
+The default config is available at [`gsm8k-llama2.yaml`](https://github.com/stanford-crfm/levanter/blob/main/examples/gsm8k-lora/gsm8k-llama2.yaml).
 
-```python
-from transformers import AutoModelForCausalLM, AutoTokenizer
+## Running Training
 
-model = AutoModelForCausalLM.from_pretrained(path)
+
+Here's how to run the command on a TPUv3-32, but you can use any TPU or GPU you want. We'll save a merged
```

Comment:
Can the default instructions be for GPU?

Reply:
sure